docs/model-dev-guide/dtrain/dtrain-implement.rst

```diff
@@ -136,17 +144,19 @@ multi-GPU training.
  Downloading Data
```

Comment:
This seems like sub-optimal advice: don't we only need to download the data once per _node_, not once per gpu?

Reply:
@garrett361 to help with rewrite

Reply:
@garrett361 which part do we want to rewrite?

Reply:
I think it should be more like
```python
import boto3
import os

def download_data_from_s3(self):
   s3_bucket = self.context.get_data_config()["bucket"]
   download_directory = "/tmp/data"
   data_file = "data.csv"
   s3 = boto3.client("s3")
   os.makedirs(download_directory, exist_ok=True)
   filepath = os.path.join(download_directory, data_file)
   if not os.path.exists(filepath):
       s3.download_file(s3_bucket, data_file, filepath)
   return download_directory

if not self.context.distributed.local_rank:
    download_directory = self.download_data_from_s3()
    self.context.distributed.broadcast_local(download_directory)
else:
    download_directory = self.context.distribued.broadcast_local(None)
```

Seem right, @azhou-determined ?


I think we'd also want to re-write some examples which use very similar code also, so can also mark this as out of scope?

Reply:
i'll mark it as out of scope for this PR and create a new one to improve the examples, i'll transfer this comment over